lightning/src/ln/channelmanager.rs

```diff
@@ -3140,18 +3140,18 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 				let mut session_priv_bytes = [0; 32];
 				session_priv_bytes.copy_from_slice(&session_priv[..]);
 				let mut outbounds = self.pending_outbound_payments.lock().unwrap();
-				if let Some(sessions) = outbounds.get_mut(&mpp_id) {
-					if sessions.remove(&session_priv_bytes) {
-						self.pending_events.lock().unwrap().push(
-							events::Event::PaymentSent { payment_preimage }
-						);
-						if sessions.len() == 0 {
-							outbounds.remove(&mpp_id);
+				let found_payment = loop {
```

Comment:
Is `loop` needed if you always `break`? Maybe just log in the appropriate `else`.

Reply:
This avoids duplicating a long-ish log message (both `else`'s would need one). Lmk if you prefer duplicate logs or if there's a better way.

Reply:
This is the cleanest I came up with:

```rust
let found_payment = if let Some(mut sessions) = outbounds.remove(&mpp_id) {
	sessions.remove(&session_priv_bytes)
} else { false };
```

Then condition on `found_payment` to either push the event or log.

Reply:
Cleaner, thanks